irohad/pending_txs_storage/impl/pending_txs_storage_impl.cpp

```diff
@@ -64,10 +68,29 @@ PendingTransactionStorageImpl::getPendingTransactions(
   while (account_batches.batches.end() != batch_iterator
          and remaining_space >= batch_iterator->get()->transactions().size()) {
     auto &txs = batch_iterator->get()->transactions();
-    response.transactions.insert(
-        response.transactions.end(), txs.begin(), txs.end());
-    remaining_space -= txs.size();
-    ++batch_iterator;
+    // can be easily extended by adding more conditions if needed
+    if (first_tx_time or last_tx_time) {
+      auto time_predicate = [&](auto &tx) {
+        bool first_tx_time_valid =
+            first_tx_time ? first_tx_time < tx->createdTime() : true;
```

Comment:
This part of expression `first_tx_time ?` is already evaluated in `if (first_tx_time or last_tx_time)`.

And you definitely do not need to check `first_tx_time ?` for every transaction, it is known aoutisde of lambda before it is defined. 

These checks should be outside of predicate, like [pseudocode]:
```
time_predicate_f = first_tx_time ? [](tx){ return first_tx_time < tx->createdTime(); } : []{return true;};
time_predicate_l = last_tx_time ? [](tx){ return tx->createdTime() < last_tx_time); } : []{return true;};
auto time_predicate = [](tx){ return time_predicate_f && time_predicate_l; }
```
or even better
```
time_predicate = first_tx_time and last_tx_time ? 
    [](tx){ return first_tx_time < tx->createdTime() and return tx->createdTime() < last_tx_time);}
    : first_tx_time ? [](tx){ return first_tx_time < tx->createdTime() }
    : last_tx_time ? [](tx){ return tx->createdTime() < last_tx_time); }
   :  [](tx){ return true; }; //or static_cast<decltype([](tx){ return true; })>(nullptr);
```